### DISCUSSION TITLE
Output CUE expression without default evaluation?

### DISCUSSION THREAD

**Discussion by rycee** [cue-lang/cue#1182] [Q&A] [answered]
Hi,

I'm having a conundrum. Given a file `a.cue` containing

``` cue
#Defs: {
    a: 10
    b: 20
}

#Result: [
    for k, v in #Defs {
        type: k
        value: int | *v
    }
]
```

I would like to produce an output file containing

``` cue
[
    { type: "a", value: *10 | int },
    { type: "b", value: *20 | int }
]
```

but haven't figured out how to do that in an idiomatic way. The usual commands don't really output what I'm looking for.

``` console
$ cue eval a.cue -e '#Result'
[{
    type:  "a"
    value: 10
}, {
    type:  "b"
    value: 20
}]

$ cue def a.cue -e '#Result'
[ for k, v in #Defs {
	type:  k
	value: int | *v
}]
```

But I found that I can provoke the cue command to give me the expected output like so:

``` console
$ cue export --out openapi a.cue -e '#Result'
#Result: cannot use value [{type:"a",value:*10 | int},{type:"b",value:*20 | int}] (type list) as struct
```

but it doesn't exactly feel like an ideal solution 🙂 Does anybody know how to get that output in a supported manner?

--- 
Note, my actual end goal is something more like

``` cue
{ type: "a", value: *10 | int } | { type: "b", value: *20 | int }
```

Basically, I would like to use `a.cue` as a "meta schema" to create a new schema. Right now I go via the list representation and convert that to the disjunction using `sed`, any suggestion of how to improve that flow would be appreciated. For example, it would be nice if one could do something like

``` cue
x: oneOf [{ type: "a", value: *10 | int }, { type: "b", value: *20 | int }]
```

**Comment by myitcv** [comment, marked as answer]
Welcome to the CUE project, @rycee!

I think you are after the [`or()`](https://tip.cuelang.org/docs/references/spec/#or) builtin. 

Regarding the manifestation/rendering of CUE, it's a very hard problem 😄  

Is there a specific reason you wanted that output? Or does the `or()` builtin obviate that need?

**Reply by rycee** [reply]
The `or()` worked just fine, I suffered a bit of temporary confusion and for about its existence 😀

Many thanks for the help!
